### ISSUE TITLE
Links to ids don't produce Markdown links

### ISSUE THREAD

**Issue by movermeyer** [xijo/reverse_markdown#82]
When the`href` is to an HTML id, no link is generated

```ruby
link_to_external_site = '<a href="https://example.com#hallo">Hallo!</a>'
link_to_id = '<a href="#hallo">Hallo!</a>'

ReverseMarkdown.convert(link_to_external_site, github_flavored: true, tag_border: '')
=> "[Hallo!](https://example.com#hallo)"

ReverseMarkdown.convert(link_to_id, github_flavored: true, tag_border: '')
=> "Hallo!"
```

In the second case, I expected `[Hallo!](#hallo)`, not `Hallo!`.

**Comment by movermeyer** [comment]
Looking into this further, it seems that this was intended behaviour at one point:

* https://github.com/xijo/reverse_markdown/blob/cd24cc3cb64d028af9345fe88bf6e5423b5e8778/spec/assets/anchors.html#L9
* https://github.com/xijo/reverse_markdown/blob/966e3d70098b1107d7f784ea115880649bfe299c/spec/components/anchors_spec.rb#L23

I'd like to argue that this is the wrong behaviour. Markdown/CommonMark supports jump links (see [this file](https://github.com/closeio/ciso8601/blob/master/CHANGELOG.md)'s table of contents for example) and the CommonMark spec allows for any characters in the [link destination](https://spec.commonmark.org/0.29/#link-destination), [including fragments](https://spec.commonmark.org/0.29/#example-497)

**Comment by xijo** [comment]
Hi,

Thanks for reporting. Sounds right: jump marks should work as you'd expect them to. I'll have a look and get back to you alright?

Cheers,
Jo

**Comment by xijo** [comment]
Released as `1.4.0`. Thanks again!
